_posts/2020-01-08-cockpit-210.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Cockpit 210 and Cockpit-podman 12
+author: kkoukiou
+date: 2020-01-08
+tags: cockpit linux podman
+slug: cockpit-210
+category: release
+summary: Cockpit with Overview, Dashboard and Machines improvements
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly. Here are the release notes from version 210.
+
+### Overview: Add CPU utilization to usage card
+
+Display CPU usage information in the usage card of the Overview page. For detailed CPU usage information, please consult the graphs page.
+
+![Overview](/images/overview-cpu-usage.png)
+
+### Dashboard: Support SSH identity unlocking when adding new machines
+
+Cockpit now offers an inline feature to unlock SSH identities when adding a new machine to the Dashboard.
+
+![unlock-keyring](/images/unlock-keyring.png)
+
+### SElinux: Introduce an Ansible automation script
+
+SELinux modifications can reviewed and exported as Ansible playbooks. These exported playbooks can be applied to other servers.
+
+![SElinux autoscript ansible](/images/selinux-autoscript-ansible.png)
+
+### Machines: Support "bridge" type network interfaces
+
+Virtual machines can now use "bridge" networking. VMs with bridged networking have full incoming and outgoing network access on a LAN, just like a physical machine.
+
+![VM NIC bridge](/images/vm-nic-bridge.png)
+
+### Machines: Support bus type disk configuration
+
+Adding and editing VM disks now supports different bus types, such as SATA, SCSI, USB, or Virtio.
+
+Using Virtio is generally the best option, for performance reasons. Other bus types may be used if an operating system doesn't support Virtio. (For example: Windows does not support Virtio by default; additional drivers need to be installed.) Another valid reason to choose something other than Virtio is if the guest OS expects the disk to show up as another type of device.
+
+[Disk choose bus type](/images/disk-choose-bus-type.png)
+
+### Podman: Configure CPU share for system containers
+
+System containers now have a configuration option to adjust CPU shares.
+
+By default, all containers have the same proportion of CPU cycles, with a value of 1024. A container’s CPU share weighting can be changed relative to the weighting of all other running containers. For more information, please refer to [`podman run`'s documentation](https://github.com/containers/libpod/blob/master/docs/source/markdown/podman-run.1.md#options), under "cpu-shares".
```

Comment:
Is it 1024? Where do you have this info from?

Reply:
(forgot to write what I think it is and what we assume in the code)
```
 $ sudo podman run --name is_it_1024 -dit fedora bash
c9cfa73320f239f1aa07acf94f9ee926df06b4f5bc308368fcde791b8256bea4

$ sudo podman inspect --format '{{.HostConfig.CpuShares}}' is_it_1024
0
```

Reply:
That was addition from garret. I don't think we know the default value for sure, and let's not expose it here as it can vary from podman version to another.

Reply:
It's from podman's documentation, specifically podman-run.

Reply:
(BTW: If we didn't know it is 1024 by default, then the UI is most likely not as good as it should be and we need to fix it accordingly.)

Reply:
Hmm, interesting, doc says 1024, but when I run container, it has 0. I'll report this in podman and see. The UI shows as default 0, which is incorrect if the default is 1024, but we test this and the default on all suported OSes is set to 0.